tests/unit/amo/components/TestUserProfile.js

```diff
@@ -0,0 +1,315 @@
+    const { store } = dispatchSignInActions();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const username = 'i-am-not-in-the-store';
+
+    const root = renderUserProfile({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('dispatches fetchUserAccount action if username param changes', () => {
+    const { store } = dispatchSignInActions({ username: 'black-panther' });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfile({
+      params: { username: 'black-panther' },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    root.setProps({ params: { username: 'killmonger' } });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username: 'killmonger',
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount if username does not change', () => {
+    const { store } = dispatchSignInActions({ username: 'black-panther' });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfile({
+      params: { username: 'black-panther' },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    root.setProps({ params: { username: 'black-panther' } });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('renders the user avatar', () => {
+    const root = renderUserProfile();
+
+    expect(root.find('.UserProfile-user-info').shallow().find(UserAvatar))
+      .toHaveProp(
+        'user',
+        getCurrentUser(root.instance().props.store.getState().users)
+      );
+  });
+
+  it('renders a placeholder avatar while user is loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-ready' } });
+
+    expect(
+      root.find('.UserProfile-user-info').shallow().find(UserAvatar)
+    ).toHaveProp('user', undefined);
+  });
+
+  it("renders the user's name", () => {
+    const root = renderUserProfile();
+
+    expect(
+      root.find('.UserProfile-user-info').shallow().find('.UserProfile-name')
+    ).toHaveText('Matt MacTofu');
+  });
+
+  it('renders LoadingText when user name is not ready', () => {
+    const root = renderUserProfile({ params: { username: 'not-ready' } });
+
+    expect(
+      root
+        .find('.UserProfile-user-info')
+        .shallow()
+        .find('.UserProfile-name')
+        .find(LoadingText)
+    ).toHaveLength(1);
+  });
+
+  it("renders the user's username if no display name exists", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        userId: 500,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(
+      root.find('.UserProfile-user-info').shallow().find('.UserProfile-name')
```

Comment:
Doing a shallow render of `Card` isn't a good approach because it reaches into the implementation of `Card`. This might break if `Card` requires a context property in the future or if it its implementation changes in other ways. 

The `Card` component already has a well established interface that you are relying on: the `header` property. You can just reach in and render it like:

```js
const headerContent = root.find('.UserProfile-user-info').prop('header');
const header = shallow(<div>{headerContent}</div>);
expect(header.find('.UserProfile-name')).toHaveText('tofumatt');
```

This comment applies to other tests that use `root.find(...).shallow()`.